lib/src/switch.dart

```diff
@@ -96,8 +96,10 @@ class _SwitchTransformer<T> extends StreamTransformerBase<Stream<T>, T> {
       controller.onCancel = () {
         var cancels = [
           if (!outerStreamDone) outerSubscription.cancel(),
-          if (innerSubscription != null) innerSubscription.cancel(),
-        ].where((f) => f != null);
+          if (innerSubscription != null) innerSubscription!.cancel(),
```

Comment:
Since we remove nulls you could do this, 🤷 
```suggestion
         innerSubscription?.cancel(),
```

Reply:
I decided to keep this as is so that for Dart 2 we remove the 2 lines below without needing to change this.